docs: Add build-details.md doc

hkjn opened the pull request:
This piece of documentation goes into a bit more depth about
how the Armbian build process works, specifically which scripts
copy which files from host -> overlay -> Armbian filesystem,
and which scripts have which responsibilities.

~_Edit:_ This PR doesn't directly depend on https://github.com/digitalbitbox/bitbox-base/pull/56, but it does describe the Armbian build process as being dockerized, so probably would make sense to merge that PR first.~

Stadicus commented on docs/os/build-details.md:
```diff
@@ -0,0 +1,30 @@
+layout: default
+title: Build details
+parent: Operating System
+nav_order: 210
```
nav_order should be between 310 (Building the Armbian base image) and 320 (Configuration), so maybe 315?

hkjn replied:
Done.

Stadicus commented on docs/os/build-details.md:
```diff
@@ -0,0 +1,30 @@
+
+This document goes into more detail on how the Armbian build process works.
+
+The [`armbian/build.sh`](https://github.com/digitalbitbox/bitbox-base/blob/master/armbian/build.sh) script is the entrypoint to the Armbian build process.
```
To have a link from the previous chapter (Building the Armbian base image) that only references the `make` command, we could note that `make` calls the `build.sh` script.

hkjn replied:
Sure, added.

hkjn commented:
Thanks for the review, @Stadicus!

I've force-pushed https://github.com/digitalbitbox/bitbox-base/commit/7b1a3ad5dab6bcafeff624428fb952e9459fe818 which addresses your review comments, PTAL.

Stadicus commented:
ACK
